opm/output/eclipse/EclipseReader.cpp

```diff
@@ -17,230 +17,163 @@
-                    SimulationConfigConstPtr sim_config = eclipseState->getSimulationConfig();
-                    restoreRSandRV(file_type, sim_config, simulator_state );
-                } 
-            } else {
-                std::string error_str = "Restart file " +  restart_filename + " does not contain data for report step " + std::to_string(reportstep) + "!\n";
-                throw std::runtime_error(error_str);
-            }
-            ecl_file_close(file_type);
-        } else {
-            std::string error_str = "Restart file " + restart_filename + " not found!\n";
-            throw std::runtime_error(error_str);
-        }
+        return sol;
     }
 
-
-    static void restoreOPM_XWELKeyword(const std::string& restart_filename, int reportstep, bool unified, WellState& wellstate)
-    {
-        const char * keyword = "OPM_XWEL";
-        const char* filename = restart_filename.c_str();
-        ecl_file_type* file_type = ecl_file_open(filename, 0);
-
-        if (file_type != NULL) {
-
-            bool block_selected = unified ? ecl_file_select_rstblock_report_step(file_type , reportstep) : true;
-
-            if (block_selected) {
-                ecl_kw_type* xwel = ecl_file_iget_named_kw(file_type , keyword, 0);
-                const double* xwel_data = ecl_kw_get_double_ptr(xwel);
-                std::copy_n(xwel_data + wellstate.getRestartTemperatureOffset(), wellstate.temperature().size(), wellstate.temperature().begin());
-                std::copy_n(xwel_data + wellstate.getRestartBhpOffset(), wellstate.bhp().size(), wellstate.bhp().begin());
-                std::copy_n(xwel_data + wellstate.getRestartPerfPressOffset(), wellstate.perfPress().size(), wellstate.perfPress().begin());
-                std::copy_n(xwel_data + wellstate.getRestartPerfRatesOffset(), wellstate.perfRates().size(), wellstate.perfRates().begin());
-                std::copy_n(xwel_data + wellstate.getRestartWellRatesOffset(), wellstate.wellRates().size(), wellstate.wellRates().begin());
-            } else {
-                std::string error_str = "Restart file " +  restart_filename + " does not contain data for report step " + std::to_string(reportstep) + "!\n";
-                throw std::runtime_error(error_str);
-            }
-            ecl_file_close(file_type);
-        } else {
-            std::string error_str = "Restart file " + restart_filename + " not found!\n";
-            throw std::runtime_error(error_str);
-        }
+    inline data::Wells restoreOPM_XWEL( ecl_file_type* file,
+                                        int num_wells,
+                                        int num_phases ) {
+        const char* keyword = "OPM_XWEL";
+
+        ecl_kw_type* xwel = ecl_file_iget_named_kw( file, keyword, 0 );
+        const double* xwel_data = ecl_kw_get_double_ptr(xwel);
+        const double* xwel_end  = xwel_data + ecl_kw_get_size( xwel );
+
+        const double* bhp_begin = xwel_data;
```

Comment:
> They're iterators in this context

They are used as iterators - but surely pointer arithmetic is used; as long as it is tested it is fine by my though.